crates/red_knot/src/semantic/types/infer.rs

```diff
@@ -184,6 +234,57 @@ pub fn infer_definition_type(
     }
 }
 
+/// Return the type that the given constraint (an expression from a control-flow test) requires the
+/// given symbol to have. For example, returns ~None as the constraint type if given the symbol ID
+/// for x and the expression ID for `x is not None`. Returns None if the given expression applies
+/// no constraints on the given symbol.
+#[tracing::instrument(level = "trace", skip(db))]
```

Comment:
```suggestion
/// for `x` and the expression ID for `x is not None`. Returns `None` if the given expression applies
/// no constraints on the given symbol.
#[tracing::instrument(level = "trace", skip(db))]
```

Reply:
The section here is a bit hard to read because it's hard to tell where `None` refers to Rust's `None` and where it refers to `Type::None`.

Reply:
> The section here is a bit hard to read because it's hard to tell where None refers to Rust's None and where it refers to Type::None

Hmm, how can it ever be ambiguous? Bare `None` always has to be Rust `None`; `Type::None` always must have the `Type::` prefix.

But once we have typeshed I expect this will all change anyway, because `Type::None` should go away in favor of a `Type::Instance(<builtin NoneType class>)`

Reply:
Oh, I see, you mean the comment. I assumed the same name scoping rules applied there too. Will see if I can make it more clear.

Reply:
Adjusted the doc string text to be clearer.